apps/docs/app/ui/components/propTable/PropTableRender.tsx

```diff
@@ -0,0 +1,112 @@
+"use client";
+
+import {
+    flexRender,
+    getCoreRowModel,
+    useReactTable,
+    type ColumnDef
+} from "@tanstack/react-table";
+import clsx from "clsx";
+
+import type { ReactNode } from "react";
+
+export interface Item {
+    id: string;
+    name: ReactNode;
+    type: ReactNode;
+    defaultValue: string;
+    description: ReactNode;
+}
+
+const columns: ColumnDef<Item>[] = [
+    {
+        id: "name",
+        accessorKey: "name",
+        header: () => "Prop",
+        cell: info => info.getValue()
+    },
+    {
+        id: "type",
+        accessorKey: "type",
+        header: () => "Type",
+        cell: info => info.getValue()
+    },
+    {
+        id: "default",
+        accessorKey: "defaultValue",
+        header: () => "Default"
+
+    },
+    {
+        id: "description",
+        accessorKey: "description",
+        header: () => "Description",
+        cell: info => info.getValue()
+    }
+];
+
+const isColumnAvailable = (columnName: string, items: Item[]) => {
+    return items.some(item => item[columnName as keyof Item] !== "");
+};
+
+export const PropTableRender = ({ items }: { items: Item[] }) => {
+    const table = useReactTable({
+        columns,
+        state: {
+            columnVisibility: {
+                "default": isColumnAvailable("defaultValue", items)
+            }
+        },
+        data: items,
+        getCoreRowModel: getCoreRowModel()
+    });
+
+
+    return (
+        <table className="hd-table hd-props-table">
+            <thead className="hd-props-table__thead">
+                {table.getHeaderGroups().map(headerGroup => (
+                    <tr className="hd-table__column" key={headerGroup.id}>
+                        {headerGroup.headers.map(header => (
+                            <th key={header.id} className="hd-table__th">
+                                {header.isPlaceholder
+                                    ? null
+                                    : flexRender(
+                                        header.column.columnDef.header,
+                                        header.getContext()
+                                    )}
+                            </th>
+                        ))}
+                    </tr>
+                ))}
+            </thead>
+            <tbody className="hd-props-table__tbody">
+                {table.getRowModel().rows.map(row => (
+                    <tr key={row.id} className="hd-table__row hd-props-table__row">
+                        {row.getVisibleCells().map(cell => {
+                            let headerValue = "";
+                            if (typeof cell.column.columnDef.header === "function") {
+                                // TODO find a correct way to get the header value
```

Comment:
is this an expected TODO, or something you forgot?

Reply:
it's planned, I'll work on it in the next PR